ironfish/src/network/peers/peerManager.ts

```diff
@@ -782,55 +793,54 @@ export class PeerManager {
     }
   }
 
-  start(): void {
-    this.distributePeerListHandle = setInterval(() => this.distributePeerList(), 5000)
-    this.disposePeersHandle = setInterval(() => this.disposePeers(), 2000)
+  async start(): Promise<void> {
+    await Promise.allSettled([
+      (this.requestPeerListHandle = setInterval(() => this.requestPeerList(), 60000)),
+      (this.disposePeersHandle = setInterval(() => this.disposePeers(), 2000)),
+      // eslint-disable-next-line @typescript-eslint/no-misused-promises
+      (this.savePeerAddressesHandle = setInterval(async () => {
+        await this.addressManager.save(this.peers)
+      }, 60000)),
+    ])
   }
 
   /**
    * Call when shutting down the PeerManager to clean up
    * outstanding connections.
    */
-  stop(): void {
-    this.distributePeerListHandle && clearInterval(this.distributePeerListHandle)
+  async stop(): Promise<void> {
+    this.requestPeerListHandle && clearInterval(this.requestPeerListHandle)
     this.disposePeersHandle && clearInterval(this.disposePeersHandle)
+    this.savePeerAddressesHandle && clearInterval(this.savePeerAddressesHandle)
+    await this.addressManager.save(this.peers)
     for (const peer of this.peers) {
       this.disconnect(peer, DisconnectingReason.ShuttingDown, 0)
     }
   }
 
-  private distributePeerList() {
-    const connectedPeers = []
-
-    for (const p of this.identifiedPeers.values()) {
-      if (p.state.type !== 'CONNECTED') {
-        continue
-      }
-
-      connectedPeers.push({
-        identity: p.state.identity,
-        name: p.name || undefined,
-        address: p.address,
-        port: p.port,
-      })
-    }
-
-    const peerList: PeerList = {
-      type: InternalMessageType.peerList,
-      payload: { connectedPeers },
-    }
-
+  private requestPeerList() {
     const peerListRequest: PeerListRequest = {
       type: InternalMessageType.peerListRequest,
     }
 
     for (const peer of this.getConnectedPeers()) {
-      if (peer.version !== null && peer.version >= MIN_VERSION_FOR_PEER_LIST_REQUESTS) {
-        peer.send(peerListRequest)
-        continue
-      }
+      peer.send(peerListRequest)
+    }
+  }
 
-      peer.send(peerList)
+  /**
+   * Gets a random disconnected peer address and returns a peer created from
+   * said address
+   */
+  getRandomDisconnectedPeer(): Peer | null {
+    const peerAddress = this.addressManager.getRandomDisconnectedPeerAddress(this.peers)
+    if (peerAddress) {
+      const peer = this.getOrCreatePeer(peerAddress.identity)
+      peer.setWebSocketAddress(peerAddress.address, peerAddress.port)
+      peer.name = peerAddress.name || null
+      return peer
+    } else {
+      return null
```

Comment:
This is purely my opinion but I see this done a lot, and I'm kind of convinced looking at other code bases that it's simpler to have less nesting and have the main use case at the end.

```suggestion
    if (!peerAddress) {
          return null
    }
    
    const peer = this.getOrCreatePeer(peerAddress.identity)
    peer.setWebSocketAddress(peerAddress.address, peerAddress.port)
    peer.name = peerAddress.name || null
    return peer
```

Reply:
Yeah, that makes sense. I'll make the change.